### DISCUSSION TITLE
Add Custom Holiday Rules to a calendar

### DISCUSSION THREAD

**Discussion by oliverm91** [vacanza/holidays#1856] [Ideas]
In my country (Chile, CL) all legal holidays work well, but financial entities have one extra holiday (31 of december, aka "Feriado Bancario" or _roughly_ "Banking holiday"). It would be great to have the ability of adding extra holiday rules as this one which would be an `MonthDayRule` or `CustomMonthDayRule` for example.

**Comment by KJhellico** [comment]
Hi, @oliverm91

I didn't exactly understand what you meant by holiday rules, but thanks for your suggestion about Chile bank holidays. I made PR #1857, and they will be supported by Python Holidays (in BANK category).

**Comment by arkid15r** [comment]
Here is a [link](https://python-holidays.readthedocs.io/en/latest/examples.html#holiday-categories-support) to holiday categories documentation.

As @KJhellico mentioned we don't use any rules abstractions. Instead the holidays list is populated directly based on related conditions.

Please let us know if this answers your questions.


Thank you

**Comment by oliverm91** [comment]
@KJhellico @arkid15r This update fixed my issue. Thanks for the fast implementation.
